README.md

```diff
@@ -1,18 +1,120 @@
+
+- staticATokens will be transfered from the user account to the L1 bridge
+- A `deposit` event will be emitted with the L1 token address, the recipient address on L2, and the amount
+- A message will be sent to the  L2 bridge with the amount to be transferred, the L1 token address and the recipient address as parameters.
+- The token bridge on L2 will then be minting the correspending ETHStaticAToken of the L1 token to the user.
+
+Calling `depositUnderlying` will result in the following:
+
+Users will have to approve the bridge to spend the underlying `asset` tokens or `aTokens`, depending on the provided value for `fromAsset` argument when depositing.
+
+- the underlying asset of the staticAToken or the aToken (depending on the user's input )  will be transfered from the user account to the L1 bridge
+- the bridge will then deposit the aTokens/asset on the staticAToken
+- new staticATokens will be minted to the bridge
+- A message will be sent to the  L2 bridge with the amount to be transferred, the L1 token address and the recipient address as parameters.
+- The token bridge on L2 will then be minting the correspending ETHStaticAToken of the L1 token to the user.
+
+
+### Transfer L2->L1:
+
+To bridge their staticATokens back to L1, users need to call `initiate_withdraw()` on Starknet `token_bridge`. 
+
+Calling `initiate_withdraw` will result in the following:
+
+- The amount to withdraw will be burned by the bridge
+- A message will be sent to L1 with the L1 token address, the l1 recipient, the l2 rewards index and the amount
+- The L1 bridge will then transfer the staticATokens to the l1 recipient
+- The L1 bridge also checks for any difference in the l1/l2 rewards index and transfers any unclaimed rewards to the l1 user
+
+
+## Synchronisation of rewards on L1 <> L2
+
+Starknet users will continue to enjoy the same rewards as on L1 after bridging their assets; To achieve that we continously update the `rewards_index` of all ETHStaticATokens to match the value of their respective StaticATokens on L1, by tracking the reward index on departure of the static token and sending the rewards accrued during the bridging process to the recipients address.
+
+
+## Claiming rewards on L2
+
+
+To claim rewards users need to call `claim_rewards()` on ETHStaticAToken contract. The ETHStaticAToken will then be calling the bridge to mint the due `rewAAVE` tokens to the user.
+
+## Bridging rewards from L2->L1
+
+To bridge their rewards to L1, users will have to call `bridge_rewards()` on the L2 `token_bridge` providing the `amount` to be bridged. The L1 bridge will be claiming pending rewards to self from all staticATokens in a loop until having enough rewards balance to transfer it to users. The claiming ans transferring of rewards is handled by `receiveRewards()` on `TokenBridge.sol`
+
+
+## Proxies
+
+All calls made to the following contracts will be handled by a proxy who delegates the calls to the available implementation of these contracts.
```

Comment:
I think other contracts should be added to your list. Almost all contracts are behind proxies now